### PR TITLE
[nrf noup] include: zephyr: Add enum for Wi-Fi rate flags

### PR DESCRIPTION
**Pull request by D-Triveni**
Introduce an enum to manage Wi-Fi rate flags.

### CODE REVIEW

File: include/zephyr/net/wifi.h [hunk 1]
```diff
@@ -30,6 +30,21 @@
...
 #endif
+/**
+ * @brief - Wi-Fi rate flags
+ */
+enum wifi_rate_flags {
+	/* Legacy */
+	LEGACY = 0,
+	/* High Throuput (11n) */
+	FORMAT_11N,
+	/* Very hight throughput(11ac) */
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:41]:
```suggestion
	/* Very high throughput (11ac) */
```

File: include/zephyr/net/wifi.h [hunk 2]
```diff
@@ -30,6 +30,21 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
+/**
+ * @brief - Wi-Fi rate flags
+ */
+enum wifi_rate_flags {
+	/* Legacy */
+	LEGACY = 0,
+	/* High Throuput (11n) */
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:39]:
```suggestion
	/* High Throughput (11n) */
```

File: include/zephyr/net/wifi.h [hunk 3]
```diff
@@ -30,6 +30,21 @@
...
+	FORMAT_11N,
+	/* Very hight throughput(11ac) */
+	FORMAT_VHT,
+	/* HE SU */
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:43]:
```suggestion
	/* 802.11ax single user */
```

File: include/zephyr/net/wifi.h [hunk 4]
```diff
@@ -30,6 +30,21 @@
...
+	HE_SU,
+	/** HE ER SU */
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:45]:
```suggestion
	/** 802.11ax emergency single user */
```

File: include/zephyr/net/wifi.h [hunk 5]
```diff
@@ -30,6 +30,21 @@
 #ifdef __cplusplus
 extern "C" {
 #endif
+/**
+ * @brief - Wi-Fi rate flags
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:34]:
```suggestion
 * @brief - Wi-Fi rate flags: These are per-packet flags that decide how the packet needs to be transmitted, the selected mode helps in decoding other parameters like MCS, data rates, etc.
```

File: include/zephyr/net/wifi.h [hunk 6]
```diff
@@ -30,6 +30,21 @@
...
+ */
+enum wifi_rate_flags {
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:36]: If this is only used for TX injection, then we should prefix it with `tx_injection`

**Review by krish2718 — CHANGES_REQUESTED** [review]
Having `noup` PRs in Zephyr should be the last resort esp. for functional code, either upstream this or move to a public header file in `sdk-nrf`.

**Comment by D-Triveni** [comment]
As per offline discussion, closing this PR and uisng the existing enum from osal files.

**Pull request closed without merging**